crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -72,22 +67,177 @@ mod advancing_time_tests {
             .await
             .unwrap();
 
-        let new_contract_address = get_udc_deployed_address(
+        get_udc_deployed_address(
             FieldElement::ZERO,
             declaration_result.class_hash,
             &starknet_rs_core::utils::UdcUniqueness::NotUnique,
             &[],
+        )
+    }
+
+    pub async fn get_current_timestamp(
+        devnet: &BackgroundDevnet,
+        contract_address: FieldElement,
+    ) -> FieldElement {
+        let call_current_timestamp = FunctionCall {
+            contract_address,
+            entry_point_selector: get_selector_from_name("get_timestamp").unwrap(),
+            calldata: vec![],
+        };
+        devnet
+            .json_rpc_client
+            .call(call_current_timestamp.clone(), BlockId::Tag(BlockTag::Latest))
+            .await
+            .unwrap()[0]
+    }
+
+    #[tokio::test]
+    async fn timestamp_syscall_set_in_past() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let new_contract_address = setup_timestamp_contract(&devnet).await;
+
+        // set time in past
+        let past_time = 1;
+        let set_time_body = Body::from(json!({ "time": past_time }).to_string());
+        let resp_set_time = devnet.post_json("/set_time".into(), set_time_body).await.unwrap();
+        let resp_body_set_time = get_json_body(resp_set_time).await;
+        assert_eq!(resp_body_set_time["block_timestamp"], past_time);
+
+        // mine block
+        devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        // check if timestamp is greater/equal
+        let current_timestamp = get_current_timestamp(&devnet, new_contract_address).await;
+        assert_ge_with_buffer(current_timestamp.to_string().parse::<u64>().ok(), Some(past_time));
```

Comment:
Here and in other places, wouldn't it be better to let `get_current_timestamp` do the conversion from `FieldElement` to `Option<u64>`, rather than doing it in multiple places.

Reply:
fixed in 5b8dba1ad980cd3310fd4a14ed78eb636e12b8ea